langwatch/prisma/migrations/20260728120003_vk_trace_project_and_managed_budget/migration.sql

```
@@ -0,0 +1,19 @@
+-- the governance project, exactly as before.
+--
+-- GatewayBudget.managedByVirtualKeyId: marks the one budget row the
+-- virtual-key drawer's budget field manages. Clearing the field archives
+-- only this row; independently created budgets keep their own lifecycle
+-- and permission boundary. No backfill: pre-existing key-targeted budgets
+-- stay unmanaged and are never archived implicitly.
+--
+-- Both additive and nullable; the rollback is dropping the columns.
+ALTER TABLE "VirtualKey" ADD COLUMN "traceProjectId" TEXT;
+ALTER TABLE "GatewayBudget" ADD COLUMN "managedByVirtualKeyId" TEXT;
+CREATE INDEX "GatewayBudget_managedByVirtualKeyId_idx" ON "GatewayBudget"("managedByVirtualKeyId");
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)